projects/storefrontlib/src/cms-structure/services/feature-modules.service.ts

```diff
@@ -0,0 +1,286 @@
+import {
+  Compiler,
+  Injectable,
+  InjectFlags,
+  Injector,
+  NgModuleFactory,
+  NgModuleRef,
+  OnDestroy,
+} from '@angular/core';
+import {
+  CMSComponentConfig,
+  CmsComponentMapping,
+  CmsConfig,
+  ConfigChunk,
+  ConfigInitializerService,
+  configurationFactory,
+  DefaultConfigChunk,
+  FeatureModuleConfig,
+} from '@spartacus/core';
+import {
+  combineLatest,
+  defer,
+  forkJoin,
+  from,
+  merge,
+  Observable,
+  of,
+  queueScheduler,
+} from 'rxjs';
+import {
+  map,
+  observeOn,
+  pluck,
+  shareReplay,
+  switchMap,
+  tap,
+} from 'rxjs/operators';
+
+interface FeatureInstance extends FeatureModuleConfig {
+  moduleRef?: NgModuleRef<any>;
+  depsModules?: any[];
+  componentsMappings?: CMSComponentConfig;
+}
+
+/**
+ * Service responsible for resolving cms config based feature modules.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class FeatureModulesService implements OnDestroy {
+  // feature modules configuration
+  private featureModulesConfig?: {
+    [featureName: string]: FeatureModuleConfig;
+  };
+
+  // maps componentType to feature
+  private componentFeatureMap: Map<string, string> = new Map();
+
+  /*
+   * Contains either FeatureInstance or FeatureInstance resolver for not yet
+   * resolved feature modules
+   */
+  private features: Map<string, Observable<FeatureInstance>> = new Map();
+
+  private dependencyModules = new Map<any, NgModuleRef<any>>();
+
+  constructor(
+    protected configInitializer: ConfigInitializerService,
+    protected compiler: Compiler,
+    protected injector: Injector
+  ) {
+    this.initFeatureMap();
+  }
+
+  private async initFeatureMap(): Promise<void> {
+    const config: CmsConfig = await this.configInitializer.getStableConfig(
+      'featureModules'
+    );
+
+    this.featureModulesConfig = config.featureModules ?? {};
+
+    for (const [featureName, featureConfig] of Object.entries(
+      this.featureModulesConfig
+    )) {
+      if (featureConfig.cmsComponents?.length) {
```

Comment:
Edge case with disabling a feature:
```
featureModules: {
   storeFinder: null
}
```

```suggestion
      if (featureConfig?.cmsComponents?.length) {
```